packages/js/src/plugins/tokenModule/TokenClient.ts

```diff
@@ -45,10 +45,31 @@ import {
   sendTokensOperation,
   SendTokensOutput,
 } from './sendTokens';
+import {
+  AddMetadataInput,
+  addMetadataOperation,
+  AddMetadataOutput,
+} from './addMetadata';
+import { MintWithMetadata } from '@/plugins';
 
 export class TokenClient {
   constructor(protected readonly metaplex: Metaplex) {}
 
+  addMetadata(
+    input: AddMetadataInput
+  ): Task<AddMetadataOutput & { mintWithMetadata: MintWithMetadata | Mint }> {
```

Comment:
If you're adding metadata and you end up with a `Mint` model then something went wrong.

```suggestion
  ): Task<AddMetadataOutput & { mintWithMetadata: MintWithMetadata }> {
```

You can assert that `mintWithMetadata` is not a `Mint` model by using `assertMintWithMetadata(...)`.